Source/HTTP/XCurl/XCurlProvider.cpp

```diff
@@ -0,0 +1,128 @@
+#include "pch.h"
+#include "XCurlProvider.h"
+#include "XCurlEasyRequest.h"
+
+using namespace xbox::http_client;
+
+HRESULT Internal_InitializeHttpPlatform(HCInitArgs* args, PerformEnv& performEnv) noexcept
+{
+    assert(args == nullptr);
+    UNREFERENCED_PARAMETER(args);
+
+    auto initResult = HC_PERFORM_ENV::Initialize();
+    RETURN_IF_FAILED(initResult.hr);
+
+    performEnv = initResult.ExtractPayload();
+
+    return S_OK;
+}
+
+void Internal_CleanupHttpPlatform(HC_PERFORM_ENV* performEnv) noexcept
+{
+    // HC_PERFORM_ENV created with custom deleter - ~HC_PERFORM_ENV needs to be explicitly invoked.
+    performEnv->~HC_PERFORM_ENV();
```

Comment:
do we need to actually delete the memory?

Reply:
good question.  Will look at other implementations and test

Reply:
added explicit deallocation